### DISCUSSION TITLE
Question about ARM devices

### DISCUSSION THREAD

**Discussion by diogoleon1** [skyline-emu/skyline#1215] [Q&A] [answered]
Do you plan to optimize the emulator to run games natively as android apps?

**Comment by yhak22** [comment]
It's impossible because the emulation depends on the spec of the cpu and gpu 
You need high end phone for beter emulation.

**Reply by StringLibrary** [reply]
To be honest it is seen that you speak Spanish and you are native but no matter your question could be answered like this or that I think you want the games not to be emulated but to run natively first to achieve that the developers of the emulator should no longer emulate anything but work directly with the GPU and CPU probably if someone achieves it with the passage of time your cell phone would suffer severe traumas since they are games made for switch not for a Phones but if the companies that create the cell phones are aligned with those that create the video game consoles there if I can believe that that happens but it looks doubtful and if this emulator works directly with the GPU and CPU they would have to do something very complicated From makeup, tricking video games to work and trying to lower the resolution there come a lot of things but now you understand me or not?

**Reply by yhak22** [reply]
Oh , bro 
I think you think that I'm the one who post this question but no u ask the wrong person  
the one who ask is on top ,
I simply answered his question , and I'm not Spanish though.

**Comment by PixelyIon** [comment, marked as answer]
I'm not sure what exactly you mean but if you mean skipping over CPU emulation then we already do that with NCE (Native Code Execution) but if you mean modifying the game code to optimize it further then we can't really do that without mods.
